src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -54,17 +54,15 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
                     chain.ChainPolicy.ExtraStore.AddRange(additionalCertificates);
                 }
 
-                if (trust != null)
+                if (trust?._store?.Certificates.Count > 0)
```

Comment:
The `Certificates` property on X509Store reads the store live and instantiates all the certificates.  Using it in this manner throws a bunch of data at the GC and finalizer queue.  And has a ToC/ToU bug, as it could still end up as empty on line 60.